apps/web/src/actions/documents/create.ts

```diff
@@ -19,6 +25,13 @@ export const createDocumentVersionAction = authProcedure
       name: z.string(),
       parentId: z.number().optional(),
     }),
-    { type: 'json' },
   )
-  .handler(async ({ input }) => createDocumentVersion(input))
+  .handler(async ({ input }) => {
+    await createDocumentVersion(input)
+    await materializeDocumentsAtCommit({
+      commitUuid: input.commitUuid,
+      staged: true,
+    })
+
+    revalidatePath(ROUTES.root)
```

Comment:
can we keep it a regular pure function that executes a write and returns the result? In general actions that perform side effects are more complex to maintain.

Reply:
Yes agree. I was just trying someting